lib/puppet/type/cisco_vtp.rb

```diff
@@ -74,10 +74,17 @@
   newproperty(:filename) do
     desc "VTP file name. Valid values are string, keyword 'default'."
 
-    munge do |file_name|
-      file_name = :default if file_name == 'default'
+    validate do |file_name|
       fail 'File name is not a string.' unless
         file_name == :default || file_name.is_a?(String)
+    end
+
+    munge do |file_name|
+      file_name = :default if file_name == 'default'
+      unless file_name == :default || /(bootflash:|usb.*:)/.match(file_name)
```

Comment:
Nit: _maybe_ a little cleaner to do:

```
# Prefix bootflash:/ if only given a filename
file_name.insert(0, 'bootflash:/') unless file_name[/\//]

```

...but what you have is fine too. :+1: either way